workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlows.cs

```diff
@@ -0,0 +1,289 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Improbable.Worker.CInterop;
+using Improbable.Worker.CInterop.Alpha;
+using Locator = Improbable.Worker.CInterop.Locator;
+using LocatorParameters = Improbable.Worker.CInterop.LocatorParameters;
+using AlphaLocator = Improbable.Worker.CInterop.Alpha.Locator;
+
+namespace Improbable.Gdk.Core
+{
+    public interface IConnectionFlow
+    {
+        Task<Connection> CreateAsync(ConnectionParameters parameters, CancellationToken? token = null);
+    }
+
+    public class ReceptionistFlow : IConnectionFlow
+    {
+        public string ReceptionistHost = RuntimeConfigDefaults.ReceptionistHost;
+        public ushort ReceptionistPort = RuntimeConfigDefaults.ReceptionistPort;
+        public string WorkerId;
+
+        public ReceptionistFlow(string workerId, IConnectionFlowInitializer<ReceptionistFlow> initializer = null)
+        {
+            WorkerId = workerId;
+            initializer?.Initialize(this);
+        }
+
+        public async Task<Connection> CreateAsync(ConnectionParameters parameters, CancellationToken? token = null)
+        {
+            using (var connectionFuture =
+                Connection.ConnectAsync(ReceptionistHost, ReceptionistPort, WorkerId, parameters))
+            {
+                return await Utils.TryToConnectAsync(connectionFuture, token).ConfigureAwait(false);
+            }
+        }
+    }
+
+    public class LocatorFlow : IConnectionFlow
+    {
+        public string LocatorHost = RuntimeConfigNames.LocatorHost;
+
+        public LocatorParameters LocatorParameters = new LocatorParameters
+        {
+            LoginToken = new LoginTokenCredentials
+            {
+                Token = string.Empty
+            },
+            Steam = new SteamCredentials
+            {
+                Ticket = string.Empty,
+                DeploymentTag = string.Empty
+            },
+            Logging = new ProtocolLoggingParameters(),
+            ProjectName = string.Empty,
+            CredentialsType = LocatorCredentialsType.LoginToken,
+            EnableLogging = false,
```

Comment:
do we need to explicitly set this to false? if we do here - should do the same when initialising `Alpha.LocatorParameters`

Reply:
Not strictly speaking - but we set everything to a safe default so it was just for consistency. Do you feel particularly strongly about this?